### PR TITLE
Write more testcases & allow Name to contains non-alphanumeric char

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

File: src/main/java/trackitnus/model/commons/Name.java [hunk 1]
```diff
@@ -11,13 +11,13 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-        "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+        "Names should be a non-empty string and don't contain '/'";
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/commons/Name.java]: I think should change to `Names should be a non-empty string and should not contain '/'`

File: src/main/java/trackitnus/model/lesson/LessonComparator.java [hunk 2]
```diff
@@ -6,15 +6,15 @@
     /**
      * Compares 2 lessons in chronological order to facilitate sorting.
      *
-     * @param firstLesson First lesson to compare.
+     * @param firstLesson  First lesson to compare.
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/lesson/LessonComparator.java:9]: Is the extra space supposed to be there?

**Review by simonteozw — COMMENTED** [review]
Looks good! Just a few comments

File: src/main/java/trackitnus/model/lesson/LessonComparator.java (hunk 2 above)
Comment (UncleGrandpa925) [reply src/main/java/trackitnus/model/lesson/LessonComparator.java:9]: No 😢 I think this happened during merging

File: src/main/java/trackitnus/model/commons/Name.java (hunk 1 above)
Comment (UncleGrandpa925) [reply src/main/java/trackitnus/model/commons/Name.java]: Sure!

File: src/main/java/trackitnus/model/lesson/LessonComparator.java (hunk 2 above)
Comment (UncleGrandpa925) [reply src/main/java/trackitnus/model/lesson/LessonComparator.java:9]: Ah yes, it's the formatter of IntelliJ, it auto-align the comments of params

**Review by simonteozw — APPROVED** [review]
LGTM

**Pull request merged**